ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -194,14 +205,139 @@ FileSpan addPutEntries(int count, long size, long expiresAtMs) throws IOExceptio
       } else {
         liveKeys.add(id);
       }
+
+      index.addToIndex(indexEntries, fileSpan);
+      Offset previousIndexSegmentStartOffset = index.indexes.lowerKey(endOffsetOfPrevMsg);
+      checkRolledOver(previousIndexSegmentStartOffset);
```

Comment:
not sure I get the point of this function. Why not just add to the valid size in this index segment?